api/src/modules/sourcing-record-groups/sourcing-record-group.entity.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { User } from 'modules/users/user.entity';
+import { BaseServiceResource } from 'types/resource.interface';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+
+export const sourcingRecordGroupResource: BaseServiceResource = {
+  className: 'SourcingRecordGroup',
+  name: {
+    singular: 'sourcingRecordGroup',
+    plural: 'sourcingRecordGroups',
+  },
+  entitiesAllowedAsIncludes: [],
+};
+
+@Entity()
+export class SourcingRecordGroup extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  @ApiProperty()
+  id!: string;
+
+  @Column({ nullable: true })
+  @ApiProperty()
+  title!: string;
+
+  @Column({ nullable: true })
+  @ApiPropertyOptional()
+  description?: string;
+
+  @Column({ type: 'jsonb', nullable: true })
+  @ApiPropertyOptional()
+  metadata?: JSON;
+
+  @Column({
+    type: 'timestamp',
+    default: () => 'CURRENT_TIMESTAMP',
+  })
+  @ApiProperty()
+  lastEdited!: string;
+
+  @ManyToOne(() => User, (user: User) => user.sourcingRecordGroups, {
+    eager: false,
+  })
+  @ApiProperty()
+  lastEditedUser?: User;
+
+  /**
+   * @debt Auto-assign user and make not nullable
+   */
+  @Column({ nullable: true })
+  @ApiPropertyOptional()
```

Comment:
Is this decorator necesary? Legit question

Which is the point to document `lastEditedUser` as  `@ApiProperty()` and then lastEditedUserId as `@ApiPropertyOptional()`?

Reply:
great legit question. once I have a legit answer, I'll tell you :P
I need to look into this at some point, once we double down on the API docs

Reply:
I already look it up for you 😝 
I'd say this makes no sense, because of that first `@ApiProperty()`
![image](https://user-images.githubusercontent.com/37807221/124490427-5b438180-ddb2-11eb-9218-8780054f3ef2.png)